### PR TITLE
groff 1.23.0

### PR DESCRIPTION
**Pull request by p-linnane**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
    Minitest::Assertion: Expected /BSD\ General\ Commands\ Manual/ to match "TRUE(1)                     General Commands Manual                    TRUE(1)
    
    NAME
           true -- return true value
    
    SYNOPSIS
           true
    
    DESCRIPTION
           The true utility always returns with an exit code of zero.
    
           Some  shells  may  provide a builtin true command which is identical to
           this utility.  Consult the builtin(1) manual page.
    
    SEE ALSO
           builtin(1), csh(1), false(1), sh(1)
    
    STANDARDS
           The true utility is expected to be IEEE Std 1003.2 (\"POSIX.2\") compati-
           ble.
    
    GNU                              June 9, 1993                          TRUE(1)
    ".
```

**Comment by g-branden-robinson** [comment]
The "man-db test failure" is due to an upstream change.  The output you are getting is expected.

> commit 24d9e02d42f9ce1e1f97855bc8464e080b93c7b9
> Author: G. Branden Robinson <g.branden.robinson@gmail.com>
> Date:   Tue Sep 6 18:32:36 2022 -0500
> 
>     [mdoc]: Change default "OS" from "BSD" to "GNU".
>     
>     * tmac/mdoc/doc-common: Change `doc-default-operating-system` string
>       from "BSD" to "GNU", per discussion with Ingo Schwarze.  See
>       <https://lists.gnu.org/archive/html/groff/2022-07/msg00280.html>.
>     
>     * tmac/tests/doc_smoke-test.sh: Update expectations of footer regression
>       test.
>

**Comment by g-branden-robinson** [comment]
The Minitest assertion failure also appears to be due to a deliberate change.

The _man_ package in _groff_ 1.23.0 formats the man page name in italics in headers and footers, just like in man page cross references.

This is an aspect of a new feature described in the `NEWS` file.

> o The an (man) macro package supports a new macro, `MR`, intended for
>   use by man page cross references in preference to the font style
>   alternation macros historically used.  Where before you would write
>     .BR ls (1).
>   or
>     .IR ls (1).
>   you should now write
>     .MR ls 1 .
>   (the third argument, typically used for trailing punctuation, is
>   optional).  Because the macro semantically identifies a man page, it
>   can create a clickable hyperlink ("man:ls(1)" for the above example)
>   on supporting devices.  Furthermore, a new string, `MF`, defines the
>   font to be used for setting the man page topic (the first argument to
>   `MR` and `TH`), permitting configuration by distributions, sites, and
>   users.

You may want to update your test to tell `groff` to format the man page with an additional command-line parameter `-dMF=R` to use the roman face instead of italics, as implied by the news item above.

**Comment by p-linnane** [comment]
@g-branden-robinson Thank you for the advice. I've made the changes to the tests as you recommended and will see how they do in CI.

**Comment by g-branden-robinson** [comment]
Hi @p-linnane,

I see that there are still a couple of problems.  I'll take them one at a time.

The first one is a continuing man-db failure on Linux.  I am not positive that I completely understand the test case.  I see the following in the logs.

`t::Assertion: Expected /https:\/\/savannah\.nongnu\.org\/bugs\/\?group=man\-db/ to match "GMAN(1) `

...followed by the the complete formatted text of the page.  Toward the end of that text we see:

`[https://savannah.nongnu.org/bugs/?group=man‐db\n\n2.11.2](https://savannah.nongnu.org/bugs/?group=man%E2%80%90db\n\n2.11.2)`

It looks like the regex here is not expecting a UTF-8 sequence to occur in the URL.   This can be overcome by telling man-db man(1) to use the "-Tascii" option.  That will format the page using only Unicode Basic Latin code points.

The man-db man(1) page itself appears to be using the wrong kind of hyphen in its URL which is the ultimate source of this problem--but `-Tascii` will work around it.  _groff_ 1.23.0's `NEWS` file warns about this.

>   The an (man) and doc (mdoc) macro packages no longer remap the -, ',
>   and ` input characters to Basic Latin code points on UTF-8 devices,
>   but treat them as groff normally does (and AT&T troff before it did)
>   for typesetting devices, where they become the hyphen, apostrophe or
>   right single quotation mark, and left single quotation mark,
>   respectively.  This change is expected to expose glyph usage errors in
>   man pages.  See the "PROBLEMS" file for a recipe that will conceal
>   these errors.  A better long-term approach is for man pages to adopt
>   correct input practices; the man pages groff_man_style(7),
>   groff_char(7), and man-pages(7) (subsection "Generating optimal
>   glyphs"; from the Linux man-pages project) contain such instructions.
>   Doing so also improves man page typography when formatting for PDF.

As I minor point, I'm not sure the '-' needs to be escaped in the regex pattern you're using.  (Escaping hyphens can be important in *roff source documents, but it appears that formatted _output_ is what is being matched.)

**Comment by g-branden-robinson** [comment]
Hi @p-linnane,

The second remaining issue appears to be simpler.  This "ronn" test is failing on Linux and every variety of macOS.

The test is expecting multiple blank lines after headers and before footers when man pages are formatted for terminals.  This has changed upstream in _groff_ 1.23.0 to make more economical use of terminal windows when "continuously rendering".  (Historically, _nroff_ would try to achieve half-inch margins between page body text and headers/footers when formatting man pages; this dates all the way back to when teletypewriters were used as Unix terminals.)

There is not a `NEWS` item for this; it was a consequence of many revisions to resolve [Savannah #60609](https://savannah.gnu.org/bugs/?60609) and other problems.

Long story short, the new output appears correct to me and I think you can revise the test expectations accordingly.

**Comment by g-branden-robinson** [comment]
Hi @p-linnane,

Anything I can do to help?

**Comment by chenrui333** [comment]
#136749
